scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -46,55 +60,130 @@ namespace <%= namespace %> {
         /// \endcode
         public Cart(ShopifyClient client) {
             Client = client;
-            _LineItems = new CartLineItems();
+            _LineItems = new CartLineItems(OnDeleteLineItem);
         }
 
-        /// <summary>
-        /// Returns a link which can be used to check out this cart's line items via a web browser.
-        /// </summary>
-        /// <param name="note">can be used to create a special note for the merchant for this order</param>
-        /// \code
-        /// // Example that creates a checkout link and opens the url in the Browser
-        /// string linkToCheckout = ShopifyBuy.Client().Cart().GetWebCheckoutLink();
-        ///
-        /// Application.OpenURL(linkToCheckout);
-        /// \endcode
-        public string GetWebCheckoutLink(string note = null) {
-            StringBuilder url = new StringBuilder();
-            bool hasLineItem = false;
+        public delegate void GetWebCheckoutLinkSuccess(string url);
+        public delegate void GetWebCheckoutLinkFailure(List<string> graphQLErrors, string httpError);
+
+        public void GetWebCheckoutLink(GetWebCheckoutLinkSuccess success, GetWebCheckoutLinkFailure failure) {
+            CheckoutSave((errors, httpError) => {
+                if (errors != null || httpError != null) {
+                    failure(errors, httpError);
```

Comment:
We're thinking of refactoring how errors are handled. More about is here https://github.com/Shopify/unity-buy-sdk/issues/105